File: 2024-1/ejercicios-java/clase-04/Circulo.java
class Circulo {
	private String identificacion;
	private double radio;
	private final double PI = 3.141592;

	public Circulo(String identificacion, double radio) {
		this.identificacion = identificacion;
		this.radio = radio;
	}

	public String getIdentificacion() {
		return identificacion;
	}

	public double getRadio() {
		return radio;
	}

	public void setIdentificacion(String identificacion) {
		this.identificacion = identificacion;
	}

	public void setRadio(double radio) {
		this.radio = radio;
	}

	public double area() {
		return PI * radio * radio;
	}

	public double perimetro() {
		return 2 * PI * radio;
	}
}

File: 2024-2/sesiones-de-clase/sesion-de-clase-16-3/Circulo.cpp
/* 
 * File:   Circulo.cpp
 * Author: erichuiza
 * 
 * Created on October 21, 2024, 3:38 PM
 */

#include "Circulo.h"

Circulo::Circulo() {
    radio = {};
}

Circulo::Circulo(double radio) {
    this->radio = radio;
}

Circulo::Circulo(const Circulo& orig) {
    *this = orig;
}

void Circulo::operator=(const Circulo& orig) {
    this->radio = orig.getRadio();
}

double Circulo::getRadio() const {
    return radio;
}

double Circulo::calcularArea() const {
    return 3.141516 * radio * radio;
}

double Circulo::calcularCircunferencia() const {
    return 2 * 3.141516 * radio;
}

void Circulo::imprimir(ostream& os) const {
    os << "Radio: " << getRadio() << endl;
    os << "Area: " << calcularArea() << endl;
    os << "Circunferencia: " << calcularCircunferencia() << endl;
}

Circulo::~Circulo() {
    cout << "Destructor de Circulo" << endl;
}
